packages/plugin-kafka/src/kafka.ts

```diff
@@ -0,0 +1,96 @@
+import assert = require('assert')
+import { resolve } from 'path'
+import { NotFoundError } from 'common-errors'
+import { Microfleet, PluginTypes, LoggerPlugin } from '@microfleet/core'
+import {
+  ProducerStream,
+  ConsumerStream,
+  createWriteStream,
+  createReadStream,
+} from 'node-rdkafka'
+
+import {
+  GlobalConfig,
+  TopicConfig,
+  ProducerStreamOptions,
+  ConsumerStreamOptions,
+} from './types'
+
+/**
+ * Relative priority inside the same plugin group type
+ */
+export const priority = 0
+export const name = 'kafka'
+export const type = PluginTypes.transport
+
+/**
+ * Defines service extension
+ */
+export interface KafkaPlugin {
+  globalConf: GlobalConfig
+  createConsumer: (
+    streamOptions: ConsumerStreamOptions,
+    globalConf?: GlobalConfig,
+    topicConf?: TopicConfig
+  ) => ConsumerStream
+  createProducer: (
+    streamOptions: ProducerStreamOptions,
+    globalConf?: GlobalConfig,
+    topicConf?: TopicConfig
+  ) => ProducerStream
+}
+
+export class KafkaFactory implements KafkaPlugin {
+  globalConf: GlobalConfig
+  constructor(globalConf: GlobalConfig) {
+    this.globalConf = globalConf
+  }
+
+  createConsumer(
```

Comment:
Postfix both method names with *Stream, in case we'll need to provide the Flowing implementation API as well